### DISCUSSION TITLE
93. InnoDB 락 중 가장 범위가 작은 것은?

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#97] [General]
1. InnoDB 락 중 가장 범위가 작은 것은? 설명도 해주세요.
2. 레코드 락은 레코드 자체를 잠그는 걸까요?
3. 인덱스가 아예 없다면 어떻게 되나요?

**Comment by linirini** [comment]
### 1. InnoDB 락 중 가장 범위가 작은 락을 설명해주세요.
InnoDB에서 가장 범위가 작은 락은 "레코드 락"이다.
레코드 락은 특정 행만을 잠그기 때문에 다른 트랜잭션이 해당 행을 제외한 다른 행에는 접근할 수 있다.
### 2. 레코드 락은 레코드 자체를 잠그는 걸까요?
실제 테이블의 레코드가 아닌 인덱스에 락을 걸어서 다른 트랜잭션에서의 수정, 삽입, 삭제 등의 작업을 막는다.
### 3. 인덱스가 아예 없다면 어떻게 되나요?
따로 생성한 인덱스가 없는 테이블은 InnoDB가 자체적으로 생성한 클러스터 인덱스를 사용하여 락을 건다.
